web/app/components/workflow/hooks/use-nodes-interactions.ts

```diff
@@ -715,6 +715,108 @@ export const useNodesInteractions = () => {
     handleSyncWorkflowDraft()
   }, [store, handleSyncWorkflowDraft, getNodesReadOnly, t])
 
+  const handleNodeCopySelected = useCallback((): undefined | Node[] => {
+    if (getNodesReadOnly())
+      return
+
+    const {
+      setClipboardElements,
+    } = workflowStore.getState()
+
+    const {
+      getNodes,
+    } = store.getState()
+
+    const nodes = getNodes()
+    const nodesToCopy = nodes.filter(node => node.selected)
```

Comment:
We manage node selection status in node.data.selected. The reason is that node.selected is managed internally by reactflow.